src/main/java/org/dita/dost/writer/KeyrefPaser.java

```diff
@@ -232,16 +232,22 @@ public void characters(final char[] ch, final int start, final int length) throw
 
     @Override
     public void endElement(final String uri, final String localName, final String name) throws SAXException {
-        if (keyrefLevel != 0 && empty && !elemName.peek().equals(MAP_TOPICREF.localName)) {
+        if (keyrefLevel != 0 && empty) {
             // If current element is in the scope of key reference element
             // and the element is empty
             if (!validKeyref.isEmpty() && validKeyref.peek()) {
                 final Element elem = keyDef.element;
                 // Key reference is valid,
                 // need to pull matching content from the key definition
-                // If current element name doesn't equal the key reference element
-                // just grab the content from the matching element of key definition
-                if (!name.equals(elemName.peek())) {
+                // If keyref on topicref, and no topicmeta, copy topicmeta from key definition
+                if (elemName.peek().equals(MAP_TOPICREF.localName)) {
+                    final NodeList topicmetaNodes = elem.getElementsByTagName(MAP_TOPICMETA.localName);
```

Comment:
This should use `XMLUtils.getChildElement(elem, MAP_TOPICMETA)` instead, because then we would not rely on element names.